weave-js/src/components/PagePanelComponents/Home/Browse3/pages/PlaygroundPage/PlaygroundChat/useChatFunctions.tsx

```diff
@@ -0,0 +1,154 @@
+import {SetStateAction} from 'react';
+
+import {Choice, Message} from '../../ChatView/types';
+import {OptionalTraceCallSchema, PlaygroundState} from '../types';
+import {DEFAULT_SYSTEM_MESSAGE} from '../usePlaygroundState';
+type TraceCallOutput = {
+  choices?: any[];
+};
+
+export const useChatFunctions = (
+  setPlaygroundStateField: (
+    index: number,
+    field: keyof PlaygroundState,
+    value:
+      | PlaygroundState[keyof PlaygroundState]
+      | SetStateAction<PlaygroundState[keyof PlaygroundState]>
+  ) => void
+) => {
+  const deleteMessage = (
+    callIndex: number,
+    messageIndex: number,
+    responseIndexes?: number[]
+  ) => {
+    setPlaygroundStateField(callIndex, 'traceCall', prevTraceCall => {
+      const newTraceCall = clearTraceCall(
+        JSON.parse(JSON.stringify(prevTraceCall))
```

Comment:
Could you use `_.deepClone`? That would be more readable and is probably safer for various types of values